src/core/qgscoordinateutils.cpp

```diff
@@ -142,6 +142,40 @@ QString QgsCoordinateUtils::formatExtentForProject( QgsProject *project, const Q
                                           QgsCoordinateUtils::formatCoordinateForProject( project, p2, destCrs, precision ) );
 }
 
+double QgsCoordinateUtils::degreeToDecimal( const QString &string, bool *ok, bool *isEasting )
+{
+  const QString negative( QStringLiteral( "swSW" ) );
+  const QString easting( QStringLiteral( "eEwW" ) );
```

Comment:
you have lowercase characters here, but the regex itself doesn't include these. I think you should add them to the regex too